backend/de.metas.business/src/main/java/de/metas/payment/api/impl/PaymentBL.java

```diff
@@ -419,6 +429,66 @@ public boolean isMatchInvoice(final I_C_Payment payment, final I_C_Invoice invoi
 		return false;
 	}
 
+	@Override
+	public String isPaypalOrCreditCardPayment(final I_C_Payment payment, final int paypalDataSourceId, final int creditCardDataSourceId)
```

Comment:
this method and it's invocation in `de.metas.banking.process.C_Payment_UpdateOrderAndInvoiceId#doIt` is unneccesary complex. 
Also:
* if it starts with `is` it should return a boolean
* don't return a String constant, but (*if* this method is not removed), an enum